internal/controller/telemetry/collector_controller.go

```diff
@@ -199,16 +200,19 @@ func (r *CollectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
-	// NOTE: This might be revised or removed in the future, but good enough for now to avoid
-	// deploying the collector that would immediately error due to configuration errors.
-	// Might also be a good place to add a validation webhook to validate the collector spec
-	if err := otelConfigInput.ValidateConfig(); err != nil {
+	// NOTE: Might also be a good place to add a validation webhook to validate the config input
```

Comment:
This isn't relevant here I think
```suggestion
```

Reply:
However the idea is right. We need a webhook to make the ux of the validation more user and operator friendly.

Reply:
And that should be a follow up task, not in scope right now.